src/main/scala/exercises/04_Options.scala

```diff
@@ -33,17 +33,26 @@ object Options {
    * - empty:  return "empty"	        E.g. None is "empty"
    * - does not exist: 					"not existing"
    */
-  def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
-  }
+  def roomState(rooms: Map[Int, Option[String]], room: Int): String = 
+    rooms.get(room) match {
+      case None                 ⇒ "not existing"
+      case Some(None)           ⇒ "empty"
+      case Some(Some("locked")) ⇒ "not available"
+      case Some(Some(s))        ⇒ s
+    }
 
   /**
    * Calculate the total amount of people in all rooms
 	 *
    * Hint: define a helper function that computes a room's occupancy
    * to convert a possible numeric String (e.g. Some("12")) to an integer
    */
-  def totalPeopleInRooms(rooms: Map[Int, Option[String]]): Int = {
-    error("Fix me")
+  
+  def occupancy(room : Option[String]) : Int = room match {
+    case None | Some("locked") ⇒ 0
+    case Some(s)               ⇒ s.toInt
   }
```

Comment:
I often forget the case A | B syntax exists in Scala.  It feels much nicer than the Java alternative, where this is implemented with fall-through cases unless you add a `break`.  This code is much more readable, and I rarely ever do want cases to fall through, so I wonder why Java adopted this convention.